src/go/rpk/pkg/cli/cmd/redpanda/start.go

```diff
@@ -738,6 +738,46 @@ func parseAddress(addr string, defaultPort int) (*config.SocketAddress, error) {
 	}, nil
 }
 
+func parseNamedAddresses(
+	addrs []string, defaultPort int,
+) ([]config.NamedSocketAddress, error) {
+	as := make([]config.NamedSocketAddress, 0, len(addrs))
+	for _, addr := range addrs {
+		a, err := parseNamedAddress(addr, defaultPort)
+		if err != nil {
+			return nil, err
+		}
+		if a != nil {
+			as = append(as, *a)
+		}
+	}
+	log.Info(as)
+	return as, nil
+}
+
+func parseNamedAddress(
+	addr string, defaultPort int,
+) (*config.NamedSocketAddress, error) {
+	if addr == "" {
+		return nil, nil
+	}
+	scheme, hostname, port, err := parseURL(addr)
```

Comment:
aalso kind of crazy there isn't a std parser for this.

Reply:
There is! This is just a convenience wrapper so that it returns what we need.

Reply:
got it.